_includes/tutorials/kafka-producer-application/kafka/markup/test/make-application-test.adoc

```diff
@@ -0,0 +1,38 @@
+////
+  This content file is used to describe how to add test code you developed in this tutorial.  You'll need to update the
+  text to suit your test code.
+
+
+////
+
+Next we will see why the thin wrapper class around the `KafkaProducer` makes testing easier.  The `KafkaProducerApplication` accepts an instance of the https://kafka.apache.org/25/javadoc/org/apache/kafka/clients/producer/Producer.html[Producer] **_interface_**.  The use of the interface allows us to inject any concrete type we want, including a https://kafka.apache.org/25/javadoc/org/apache/kafka/clients/producer/MockProducer.html[Mock Producer] for testing.
+
+We use a `MockProducer` because you only want to test your own code.  So you really only need to test that the producer recieves the expected records and in the expected format.  Plus since there is no broker, the tests run very fast, which becomes an important factor as the number of tests increase.
+
+
+
+
+There is only one method in `KafkaProducerApplicationTest` annotated with `@Test`, and that is `testProduce()`.  Before you create the test, let's go over a few of the key points of the test
+
+[source, java]
+----
+final List<String> records = Arrays.asList("foo#bar", "bar#foo", "baz#bar", "great-weather");
+
+records.forEach(producerApp::produce); <1>
+
+final List<KeyValue<String, String>> expectedList = Arrays.asList(KeyValue.pair("foo", "bar"),
+            KeyValue.pair("bar", "foo"),
+            KeyValue.pair("baz", "bar"),
+            KeyValue.pair("NO-KEY","great-weather")); <2>
+final List<KeyValue<String, String>> actualList = mockProducer.history().stream().map(this::toKeyValue).collect(Collectors.toList()); <3>
+
+----
+
+<1> Call the `produce` method
+<2> Build the expected list of records the producer should recieve
+<3> Using the `MockProducer.history()` method to get the records sent to the producer so the test can assert the expected records match the actual ones
```

Comment:
For consistency, should this be `Use` instead of `Using`?  (Or change the other ones)

Reply:
ack